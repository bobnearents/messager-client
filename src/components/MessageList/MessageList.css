@import url('https://fonts.googleapis.com/css?family=Merriweather');

.message-list-container {
  height:89.5vh;
  margin:0;
  background:#F9F8F2;
  display:inline-block;
  overflow: hidden;
  min-height: 0;
  min-width: 0;
}

.test {
  font-size: 2em;
  font-family:'Merriweather';
}

.message-list {
  padding:40px;
  display:inline-flex;
  flex-direction: column-reverse; 
  align-items: flex-start;
  overflow-y: scroll;
  width:100%;
  height:60vh;
  min-height:0;
  min-width: 0;
}

.shared {
  width:80%;
}

.full {
  width:100%;
}

.toggle-rooms {
  bottom:43vh;
  transform: translateY(-50%) rotate(-90deg);
  border:none;
  background: none;
  position: absolute;
}

.toggle-rooms:hover {
  box-shadow: 1px 1px 3px #888888;
}

.on {
  left:19%;
}

.off {
  left:-2.5%;
}

@media(max-width:668px) {
  .message-list {
    padding: 20px;
  }

  .message-list p {
    width:75%;
  }
  
  .shared{
    display: none;
  }

  .message-list-container {
    width:100%;
  }
}