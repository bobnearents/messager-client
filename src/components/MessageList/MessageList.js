import React, { Component } from 'react';
import MessageContext from '../../context/message-context'
import Message from '../Message/Message'
import './MessageList.css'
import TextInput from '../TextInput/TextInput'

class MessageList extends Component {
  constructor(props) {
    super(props);
    this.state = {  }
  }

  static contextType = MessageContext;

  getRoomName = (id) => {
    let name = '';
    this.context.rooms.forEach(room => {
      if (room.id === parseInt(id)) {
        name = room.name
      }
    })
    return name;
  }
  
  generateMessagesHtml = (messagesArray, room_id) => {
    return messagesArray.map((message, index) => {
        return (
          <Message
            text={message.content}
            key={index}
            user={message.nickname}
            date={message.date_created}
            mine = {(message.username === this.context.user) ? 'mine' : ''}
          />
        );
    });
  };

  render() {
  
    return (
        <MessageContext.Consumer>
          {value => {
            return (
              <div className={(value.isRoomActive) ? 'shared message-list-container' : 'full message-list-container'}>
                <button
                  className={(value.isRoomActive) ? 'toggle-rooms on' : 'toggle-rooms off'} 
                  onClick = {() => {value.toggleActiveRoom()}}>
                  {(value.isRoomActive) ? 'Close' : 'View Rooms'}
                </button>
                <h2 className = 'test'>{(this.getRoomName(value.room_id))}</h2>
                <ul className="message-list">
                  {this.generateMessagesHtml(value.messages, value.room_id)}
                </ul>
                <TextInput />
              </div>
            );
          }}
        </MessageContext.Consumer>
    );
  }
}
 
export default MessageList;
