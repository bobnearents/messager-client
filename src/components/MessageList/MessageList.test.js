import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import MessageList from "./MessageList";

describe("<MessageList />", () => {
  it("Renders without crashing", () => {
    shallow(<MessageList />);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<MessageList />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
