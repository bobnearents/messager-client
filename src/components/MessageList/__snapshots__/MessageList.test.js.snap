// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageList /> renders correctly 1`] = `
<div
  className="full message-list-container"
>
  <button
    className="toggle-rooms off"
    onClick={[Function]}
  >
    View Rooms
  </button>
  <h2
    className="test"
  >
    
  </h2>
  <ul
    className="message-list"
  />
  <form
    className="message-form"
    onSubmit={[Function]}
  >
    <input
      className="message-input"
      name="text"
      type="text"
    />
    <button
      className="message-submit-button"
      type="submit"
    >
      Enter
    </button>
  </form>
</div>
`;
